internal/alpini/netty4/alpini-netty4-base/src/main/java/com/linkedin/alpini/netty4/ssl/SslInitializer.java

```diff
@@ -290,9 +294,26 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
           _handshakesFailed.increment();
         }
         if (ctx.channel().hasAttr(SSL_HANDSHAKE_START_TS)) {
+          SslHandler handler = ctx.pipeline().get(SslHandler.class);
```

Comment:
Can't comment on white lines :(

But in the context of this function, I'm trying to recall/grok where we call fireUserEventTriggered, are we calling it manually or is it something that just trips when an stack trace gets thrown?

Reply:
I'm not sure about that. I found that this line gets logged every time a handshake succeeds/fails. This was the starting point of my investigation into this

Reply:
It is in Netty docs: https://netty.io/4.1/api/io/netty/handler/ssl/SslHandler.html
> Beside using the handshake [ChannelFuture](https://netty.io/4.1/api/io/netty/channel/ChannelFuture.html) to get notified about the completion of the handshake it's also possible to detect it by implement the [ChannelInboundHandler.userEventTriggered(ChannelHandlerContext, Object)](https://netty.io/4.1/api/io/netty/channel/ChannelInboundHandler.html#userEventTriggered-io.netty.channel.ChannelHandlerContext-java.lang.Object-) method and check for a [SslHandshakeCompletionEvent](https://netty.io/4.1/api/io/netty/handler/ssl/SslHandshakeCompletionEvent.html).